frontend/amundsen_application/static/js/components/LineageGraph/index.tsx

```diff
@@ -0,0 +1,300 @@
+
+/* eslint-disable no-underscore-dangle, no-param-reassign */
+
+import * as d3 from 'd3';
+import * as React from 'react';
+
+import { Lineage } from 'interfaces';
+
+import './styles.scss';
+
+import { LeftIcon } from 'components/SVGIcons';
+
+export interface LineageGraphProps {
+  lineage: Lineage;
+}
+
+interface LineageGraphState {}
+
+export class LineageGraph extends React.Component<
+  LineageGraphProps,
+  LineageGraphState
+> {
+  state: any = {};
+
+  private nodeRef = React.createRef<HTMLDivElement>();
+
+  drawTree = this._drawTree.bind(this);
+
+  componentDidMount() {
+    // Centering the graph
+    const dimensions = this.nodeRef.current!.getBoundingClientRect();
+    if (this.nodeRef.current) {
+      // const dimensions = this.nodeRef.current!.getBoundingClientRect();
+      this.setState({
+        nodeTranslate: {
+          x: dimensions.width / 2,
+          y: dimensions.height / 2,
+        },
+      });
+    }
+
+    const margin = { top: 20, right: 20, bottom: 20, left: 20 };
+    // Setting up the dimensions/fallback dimensions
+    const totalWidth = dimensions.width || 1280 - (margin.left + margin.right);
+    const totalHeight =
+      dimensions.height || 1024 - (margin.top + margin.bottom);
+
+    const svgContainer = d3
+      .select(this.nodeRef.current)
+      .append('svg')
+      .attr('class', 'svg-container')
+      .attr('viewBox', [0, 0, totalWidth, totalHeight]);
+
+    svgContainer
+      .append('foreignObject')
+        .attr('class', 'direction-label  upstream-label')
+        .attr('transform', `translate(${(totalWidth / 2) - 150 - margin.right}, ${margin.top})`)
+        .html('<img class="icon icon-left-arrow" /> Upstream')
+
+    svgContainer
+      .append('foreignObject')
+        .attr('class', 'direction-label downstream-label')
+        .attr('transform', `translate(${(totalWidth / 2) + margin.left}, ${margin.top})`)
+        .html('<img class="icon icon-right-arrow" /> downstream')
+
+    this.drawTree(totalWidth / 2, totalHeight, margin);
+  }
+
+  _drawTree(width: number, height: number, margin) {
+    const { lineage } = this.props;
+
+    const stratify = d3
+      .stratify()
+      .id((d) => d.key)
+      .parentId((d) => d.parent);
+
+    let uniqueIdCounter = 0;
+    const animationDuration = 500;
+
+    const treemap = d3.tree().size([height, width / 2]);
+    const upstreamRoot = d3.hierarchy(
+      stratify(lineage.upstream_entities),
+      (d) => d.children
+    );
+    const downstreamRoot = d3.hierarchy(
+      stratify(lineage.downstream_entities),
+      (d) => d.children
+    );
+
+    const root = {
+      x0: height / 2,
+      y0: 0,
+    };
+
+    const svg = d3
+      .select('svg')
+      .append('g')
+      .attr('transform', `translate(${width})`);
+
+    // Collapse after the second level
+    // upstreamRoot.children.forEach(collapse);
+    // downstreamRoot.children.forEach(collapse);
+
+    update(root);
+
+    function collapse(d) {
```

Comment:
Is there the expectation for this initial version that reopening a closed item will return the item and all of its children to the previous state prior to when it was closed?